### PR TITLE
ZEPPELIN-3347 Fix "PYTHONPATH" for spark.pyspark in branch-0.8

### PR DESCRIPTION
**Pull request by mebelousov**
### What is this PR for?
Use system PYTHONPATH in spark.pyspark interpreter, if PYTHONPATH already exists in environment variables.

### What type of PR is it?
[Bug Fix]

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-3347

### How should this be tested?
* Add some directories to PYTHONPATH
`export PYTHONPATH="/opt/python/dir1:/opt/python/dir2:$PYTHONPATH"`
* Build branch "ZEPPELIN-3347_0.8"
* Run in %spark.pyspark
```
import sys
print('\n'.join(sys.path))
```
* Check that the directories are in the output list.
Compare with the result in %python.

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: spark/interpreter/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java [hunk 1]
```diff
@@ -204,33 +204,35 @@ private Map setupPySparkEnv() throws IOException, InterpreterException {
 
     // only set PYTHONPATH in local or yarn-client mode.
     // yarn-cluster will setup PYTHONPATH automatically.
-    SparkConf conf = getSparkConf();
+    SparkConf conf = null;
+    try {
+      conf = getSparkConf();
+    } catch (InterpreterException e) {
+      throw new IOException(e);
```
Comment (felixcheung) [inline spark/interpreter/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java]: hmm, why should this be IOException and not InterpreterException?

File: spark/interpreter/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java [hunk 2]
```diff
@@ -204,33 +204,35 @@ private Map setupPySparkEnv() throws IOException, InterpreterException {
 
     // only set PYTHONPATH in local or yarn-client mode.
     // yarn-cluster will setup PYTHONPATH automatically.
-    SparkConf conf = getSparkConf();
+    SparkConf conf = null;
+    try {
+      conf = getSparkConf();
```
Comment (felixcheung) [inline spark/interpreter/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java]: what's the reason that getSparkConf can fail here?

Comment (mebelousov) [reply spark/interpreter/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java]: I'm just copied from #3374. 
I can simplify if necessary.
@zjffdu

**Comment by zjffdu** [comment]
I have merged it, @mebelousov Could you close it ?

**Pull request closed without merging**